In [5]:
import numpy as np
import scipy.stats as st
import pandas as pd

In [6]:
df=pd.read_csv(r'D:\CSEbtech(amritha)\sem5\19cse304\my application\vgsales.csv')

In [7]:
#taking into consideration of the glabal_sales attribute of the dataset
df['Global_Sales']

0        82.74
1        40.24
2        35.82
3        33.00
4        31.37
         ...  
16593     0.01
16594     0.01
16595     0.01
16596     0.01
16597     0.01
Name: Global_Sales, Length: 16598, dtype: float64

In [8]:
#16598
df

Rank                                              Name Platform  \
0          1                                        Wii Sports      Wii   
1          2                                 Super Mario Bros.      NES   
2          3                                    Mario Kart Wii      Wii   
3          4                                 Wii Sports Resort      Wii   
4          5                          Pokemon Red/Pokemon Blue       GB   
...      ...                                               ...      ...   
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year         Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  \
0      2006.0        Sports    Nintendo     41.49     29.02      3.77   
1      1985.0      Platform    Nintendo     29.08      3.58      6.81   
2      2008.0        Racing    Nintendo     15.85     12.88      3.79   
3      2009.0        Sports    Nintendo     15.75     11.01      3.28   
4      1996.0  Role-Playing    Nintendo     11.27      8.89     10.22   
...       ...           ...         ...       ...       ...       ...   
16593  2002.0      Platform       Kemco      0.01      0.00      0.00   
16594  2003.0       Shooter  Infogrames      0.01      0.00      0.00   
16595  2008.0        Racing  Activision      0.00      0.00      0.00   
16596  2010.0        Puzzle    7G//AMES      0.00      0.01      0.00   
16597  2003.0      Platform     Wanadoo      0.01      0.00      0.00   

       Other_Sales  Global_Sales  
0             8.46         82.74  
1             0.77         40.24  
2             3.31         35.82  
3             2.96         33.00  
4             1.00         31.37  
...            ...           ...  
16593         0.00          0.01  
16594         0.00          0.01  
16595         0.00          0.01  
16596         0.00          0.01  
16597         0.00          0.01  

[16598 rows x 11 columns]

In [9]:
df['Global_Sales'].mean()

0.53744065550074

In [10]:
df['Global_Sales'].std()

1.5550279355699022

In [11]:
#taking a sample of 5000 records
dfs=df['Global_Sales'].sample(5000)

In [12]:
dfs


9599     0.13
14312    0.03
2810     0.73
3757     0.53
11904    0.07
         ... 
8327     0.17
7370     0.21
8789     0.15
5403     0.34
15491    0.02
Name: Global_Sales, Length: 5000, dtype: float64

In [13]:
dfs.mean()

0.5247459999999999

In [14]:
dfs.std()

1.3727810317148903

*hypothesis testing example*
It is claimed that the average sales of games accross the globe is 0.53744065550074 millions. A sample of 9000 games has an average global sales of 0.554775555555567 millions. The standard deviation is 1.5542173356583615. At 95% confidence level, is there enough evidence to reject the claim(or)the null hypothesis? 

In [15]:
#H0 : μ = 0.5 and Ha : μ != 0.5
n = 9000
xbar = 0.554775555555567
mu = 0.5
sigma = 1.5542173356583615
c=95/100
alpha = 1-c

In [16]:
z = (xbar-mu)/(sigma/np.sqrt(n))
z

3.343461271778467

In [17]:
alpha/2

0.025000000000000022

In [18]:
z_critical = abs(st.norm.ppf(alpha/2)) 
z_critical

1.959963984540054

In [19]:
if ((-z_critical)<z < z_critical): #Right-tailed test
    print("Null hypothesis cannot be rejected")
else:
    print("Reject null hypothesis")

Reject null hypothesis
